internal/provider/catalog/catalogs_schema.go

```diff
@@ -1,29 +1,113 @@
 package catalog
 
 import (
+	"context"
+
 	"github.com/hashicorp/terraform-plugin-framework/types"
 
-	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
-)
+	schemaD "github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 
-func catalogsSchema() schema.Schema {
-	return schema.Schema{
-		Description: "The catalogs datasource show the details of all the catalogs.",
+	superschema "github.com/FrangipaneTeam/terraform-plugin-framework-superschema"
+)
 
-		Attributes: map[string]schema.Attribute{
-			"id": schema.StringAttribute{
-				Computed: true,
+func catalogsSuperSchema(_ context.Context) superschema.Schema {
+	return superschema.Schema{
+		Common: superschema.SchemaDetails{
+			MarkdownDescription: "The catalogs datasource show the details of all the catalogs.",
+		},
+		Attributes: map[string]superschema.Attribute{
+			"id": superschema.StringAttribute{
```

Comment:
why you don't use supertypes ?

Reply:
Because it is not planned to re-write the datasource